SampleGallery/Samples/SDK Insider/Gears/Gears.cs

```diff
@@ -13,126 +13,258 @@
+        {
+            // Create a visual based on the XAML object
+            var visual = ElementCompositionPreview.GetElementVisual(gear);
+            visual.Size = new Vector2((float)gear.ActualWidth, (float)gear.ActualHeight);
+            visual.AnchorPoint = new Vector2(0.5f, 0.5f);
+            this.gearVisuals.Add(visual);
+
+            this.Count++;
+
+            return visual;
+        }
+
+        private void ConfigureGearAnimation(Visual currentGear, Visual previousGear)
+        {
+            // If rotation expression is null then create an expression of a gear rotating the opposite direction
+            rotationExpression = rotationExpression ?? compositor.CreateExpressionAnimation("-previousGear.RotationAngleInDegrees");
+
+            // put in placeholder parameters
+            rotationExpression.SetReferenceParameter("previousGear", previousGear);
+
+            // Start the animation based on the Rotation Angle in Degrees.
+            currentGear.StartAnimation("RotationAngleInDegrees", rotationExpression);
         }
 
         private void EnsureGearMotor()
         {
-            if (_gearMotor == null)
+            // Start the first gear (the red one)
+            if (gearMotionScalarAnimation == null)
             {
-                _gearMotor = _compositor.CreateScalarKeyFrameAnimation();
-                var linear = _compositor.CreateLinearEasingFunction();
+                gearMotionScalarAnimation = compositor.CreateScalarKeyFrameAnimation();
+                var linear = compositor.CreateLinearEasingFunction();
 
-                _gearMotor.InsertExpressionKeyFrame(0.0f, "this.StartingValue");
-                _gearMotor.InsertExpressionKeyFrame(1.0f, "this.StartingValue + 360", linear);
+                gearMotionScalarAnimation.InsertExpressionKeyFrame(0.0f, "this.StartingValue");
+                gearMotionScalarAnimation.InsertExpressionKeyFrame(1.0f, "this.StartingValue + 360", linear);
 
-                _gearMotor.IterationBehavior = AnimationIterationBehavior.Forever;
+                gearMotionScalarAnimation.IterationBehavior = AnimationIterationBehavior.Forever;
             }
         }
 
-        private void AnimateSlow_Click(object sender, RoutedEventArgs e)
+        private void AnimateFast_Click(object sender, RoutedEventArgs e)
         {
+            // Start red gear
             EnsureGearMotor();
-            _gearMotor.Duration = TimeSpan.FromSeconds(5);
-            _gearVisuals[0].StartAnimation("RotationAngleInDegrees", _gearMotor);
+
+            // Fast....
+            gearMotionScalarAnimation.Duration = TimeSpan.FromSeconds(1);
+
+            // I'm only animating the first gear! :-O
+            gearVisuals.First().StartAnimation("RotationAngleInDegrees", gearMotionScalarAnimation);
         }
 
-        private void AnimateFast_Click(object sender, RoutedEventArgs e)
+        private void AnimateSlow_Click(object sender, RoutedEventArgs e)
```

Comment:
AnimateSlow and AnimateFast are identical except duration - can you just put the logic into a new function that takes in the duration? It might even make sense to add it to EnsureGearMotor(), which would take a duration (and would need to rename to StartGearMotor